lib/ansible/plugins/inventory/vmware_inventory.py

```diff
@@ -0,0 +1,410 @@
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: vmware_inventory
+    plugin_type: inventory
+    short_description: VMware inventory source
+    description:
+        - Get inventory hosts from VMware environment.
+        - Uses any file which ends with vmware.yml or vmware.yaml as a YAML configuration file.
+        - 'Please make sure you mention "enable_plugins = vmware_inventory" in inventory section of ansible.cfg.'
+        - The inventory_hostname is always the 'Name' and UUID of the virtual machine. UUID is added as VMware allows virtual machines with the same name.
+    extends_documentation_fragment:
+      - inventory_cache
+    requirements:
+      - "Python >= 2.7"
+      - "PyVmomi"
+      - "requests >= 2.3"
+      - "vSphere Automation SDK - For tag feature"
+      - "vCloud Suite SDK - For tag feature"
+    options:
+        hostname:
+            description: Name of vCenter or ESXi server.
+            env:
+              - name: VMWARE_SERVER
+        username:
+            description: Name of vSphere admin user.
+            env:
+              - name: VMWARE_USERNAME
+        password:
+            description: Password of vSphere admin user.
+            env:
+              - name: VMWARE_PASSWORD
+        port:
+            description: Port number used to connect to vCenter or ESXi Server.
+            default: 443
+            env:
+              - name: VMWARE_PORT
+        validate_certs:
+            description:
+            - Allows connection when SSL certificates are not valid. Set to C(false) when certificates are not trusted.
+            default: True
+            type: boolean
+        with_tags:
+            description:
+            - Include tags and associated virtual machines.
+            - Requires 'vSphere Automation SDK' and 'vCloud Suite SDK' libraries to be installed on the given controller machine.
+            - Please refer following URLs for installation steps
+            - 'https://code.vmware.com/web/sdk/65/vsphere-automation-python'
+            - 'https://code.vmware.com/web/sdk/60/vcloudsuite-python'
+            default: False
+            type: boolean
+'''
+
+EXAMPLES = '''
+    #Sample configuration file for VMware dynamic inventory
+    plugin: vmware_inventory
+    strict: False
+    hostname: 10.65.223.31
+    username: administrator@vsphere.local
+    password: Esxi@123$%
+    validate_certs: False
+    with_tags: True
+'''
+
+import ssl
+import atexit
+from ansible.errors import AnsibleError, AnsibleParserError
+
+try:
+    # requests is required for exception handling of the ConnectionError
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
+    raise AnsibleError('"requests" Python module is required for VMware dynamic inventory plugin.'
+                       ' Install using "pip install requests"')
```

Comment:
dont raise errors on import, the whole point of HAS_X is to be able to still execute ansible even if some plugin requirement is not met, same below. Return AnsibleParserError on validate_file or parse() instead.

Reply:
also not everyone can use pip to install libraries, you should leave the message generic/installer agnostic

Reply:
Done.